doozerlib/cli/get_nightlies.py

```diff
@@ -107,16 +219,277 @@ def rc_api_url(tag, arch):
     return f"{constants.RC_BASE_URL.format(arch=arch)}/api/v1/releasestream/{tag}{arch_suffix}"
 
 
-def all_nightlies_in_phase(major, minor, arch, phase):
-    # returns the build id string or None (or raise exception), one for each arch
-    # (may want to return "schema-version" also if this ever gets more complex)
-    tag = f'{major}.{minor}.0-0.nightly'
-    rc_url = f"{rc_api_url(tag, arch)}/tags"
-    if phase:
-        rc_url += f'?phase={phase}'
-    with request.urlopen(rc_url) as req:
-        data = json.loads(req.read().decode())
-    if not data["tags"]:
-        return None
-    nightlies = [(tag["name"], tag["phase"]) for tag in data["tags"]]
-    return nightlies
+async def populate_nightly_release_data(nightly: Dict[str, str]):
+    """
+    update the nightly record with new fields from the release image:
+    releaseInfo: output from `oc adm release info -o json` for the nightly pullspec
+    equivalence: an EquivalenceBase object for comparing to other nightlies
+    """
+    release_json_str, _ = await exectools.cmd_assert_async(f"oc adm release info {nightly['pullSpec']} -o=json", retries=3)
+    info = nightly["releaseInfo"] = json.loads(release_json_str)
+    nightly["equivalence"] = EquivalenceBase(info)
+
+
+# only look up the same container info once and store it here
+image_info_cache: Dict[str, Dict] = {}
+
+
+class EquivalenceBase:
```

Comment:
This looks really close to a Nightly class :D and we could use these objects directly instead of using Dicts? But not a strong opinion..